README.rst

```diff
@@ -79,26 +79,94 @@ See also other scripts in the win32/ directory:
+below).
+
+Modes of operation
+~~~~~~~~~~~~~~~~~~
+
+When it comes to the git-fat-specific configuration settings, there is
+two modes of operation.
+
+Implicit configuration (default)
+''''''''''''''''''''''''''''''''
+
+When no configuration files are passed as arguments to the ``-c`` option,
+``git-fat`` parses the following configuration files in the order they
+are listed here:
+
+1. If ``$GIT_FAT_CONFIG`` is defined and contains a colon-separated list of files,
+   ``git-fat`` parses those files, overriding previously defined settings with the
+   same name if they exist.
+2. The regular Git configuration files (see the ``FILES`` section of  ``man git-config``)
+   are parsed, overriding any previously defined configuration settings with the same name
+   if they exist.
+3. The ``.gitfat`` configuration file at the root of the repository is parsed,
+   overriding previously defined configuration settings with the same name if they exist.
+   In this case the ``.gitfat`` file is considered to be the *main configuration file*.
+
+Explicit configuration
+''''''''''''''''''''''
+
+When one or more configuration files are passed as arguments to the ``-c`` option,
+``git-fat`` processes each of those files in the order they were passed,
+overriding previously defined configuration settings with the same name if they exist.
 
-First, create a
+In this case, the last one of the files is considered to be the *main configuration file*.
+
+Interpolation
+~~~~~~~~~~~~~
+
+``git-fat`` performs interpolation on the setting values that are used by ``git-fat`` itself.
+Interpolation consists on replacing *setting names* delimited by curly braces {} with the value
+of those settings. *Setting names* are dot-separated strings with the section, subsection (if
+applicable) and key names of settings defined in the parsed configuration files.
+
+For the sake of integrity, it is possible to specify default values for interpolation
+in the *main configuration file* (and only there).
+
+::
+
+    [defaults "original-section"]
+    key1 = default-value-1
+
+When placed in the *main configurtion file*, the section above means that if -and only if- a
+setting with *setting name* ``original-section.key`` has not been defined in any configuration
+file, the interpolation of the string ``{original-section.key}`` will yield ``default-value-1``,
+otherwise it will yield the value it was given in the configuration file where it was defined.
+
+Example files
+~~~~~~~~~~~~~~
+
+``.gitattributes``
+'''''''''''''''''''
+This file is located at the root of the repository and determines which files
+get converted to ``git-fat`` files. See
 `git attributes <http://git-scm.com/book/en/Customizing-Git-Git-Attributes>`_
-file in the root of your repository. This file determines which files
-get converted to ``git-fat`` files.
+for further information.
 
 ::
 
     cat >> .gitattributes <<EOF
-    *.deb filter=fat -crlf
+    *.deb filter=kat -crlf
```

Comment:
You can actually just push an additional commit to your branch and it will automatically be added to the pull request.  Sorry I haven't had time to review this yet.